### PR TITLE
Add cli interface to sandbox limit overrides

### PR DESCRIPTION
**Pull request by Nemo157**
I went to do #2032 and realised we still had no way to do it other than editing the database directly. I think it's worth trying to expose all the semi-regular operations in the CLI so we don't need to remember how to edit the schema.

### CODE REVIEW

File: src/bin/cratesfyi.rs [hunk 1]
```diff
@@ -482,6 +491,72 @@ impl DatabaseSubcommand {
     }
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, Subcommand)]
+enum LimitsSubcommand {
+    /// Get sandbox limit overrides for a crate
+    Get { crate_name: String },
+
+    /// Get sandbox limit overrides for all crates
+    GetAll,
```
Comment (jyn514) [inline src/bin/cratesfyi.rs:500]: maybe call this "list"?

File: src/db/overrides.rs [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+                .get::<_, Option<i64>>("max_memory_bytes")
+                .map(|i| i as usize),
+            targets: row.get::<_, Option<i32>>("max_targets").map(|i| i as usize),
+            timeout: row
+                .get::<_, Option<i32>>("timeout_seconds")
+                .map(|i| Duration::from_secs(i as u64)),
+        }
+    }
+
+    pub fn save(conn: &mut Client, krate: &str, overrides: Self) -> Result<()> {
```
Comment (jyn514) [inline src/db/overrides.rs:42]: can you have this give a warning if we raise the time limit without setting `max_targets` to 1?

Comment (Nemo157) [reply src/db/overrides.rs:42]: I added a warning here:

```console
> cargo run -- database limits set serde --timeout 1h
     Running `/home/nemo157/.cargo/shared-target/debug/cratesfyi database limits set serde --timeout 1h`
2023-02-08T16:41:46.520337Z DEBUG docs_rs::db::pool: creating database pool (if this hangs, consider running `docker-compose up -d db s3`)
previous sandbox limit overrides for serde = Some(Overrides { memory: None, targets: None, timeout: Some(3600s) })
2023-02-08T16:41:46.534303Z  WARN docs_rs::db::overrides: setting `Overrides::timeout` implies a default `Overrides::targets = 1`, prefer setting this explicitly
new sandbox limit overrides for serde = Some(Overrides { memory: None, targets: None, timeout: Some(3600s) })
```

I tried also adding a `#[arg(requires = "targets")]` to `timeout` in the CLI interface, but the error that clap gives is not great:

```console
> cargo run -- database limits set serde --timeout 1h
     Running `/home/nemo157/.cargo/shared-target/debug/cratesfyi database limits set serde --timeout 1h`
error: the following required arguments were not provided:
  --targets <TARGETS>

Usage: cratesfyi database limits set --targets <TARGETS> --timeout <TIMEOUT> <CRATE_NAME>

For more information, try '--help'.
```

So I'll leave it with just the runtime warning.

**Review by jyn514 — APPROVED** [review]

File: src/db/overrides.rs [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
```
Comment (syphar) [inline src/db/overrides.rs:42]: coming from my wrongly queued crates: 

should this validate if a crate with the given name exists?

**Review by syphar — APPROVED** [review]

File: src/db/overrides.rs (hunk 3 above)
Comment (Nemo157) [reply src/db/overrides.rs:42]: Makes sense, we're unlikely to be adding overrides before a crate's first failed build. Added it as a warning.

**Review by syphar — APPROVED** [review]

**Comment by syphar** [comment]
@Nemo157 this is fine from my side now. 

Is anything missing? Or should we merge?

**Pull request merged**